pkg/beholder/client.go

```diff
@@ -0,0 +1,332 @@
+package beholder
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	otellog "go.opentelemetry.io/otel/log"
+	otelglobal "go.opentelemetry.io/otel/log/global"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/propagation"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Emitter interface {
+	// Sends message with bytes and attributes to OTel Collector
+	Emit(ctx context.Context, body []byte, attrs map[string]any) error
```

Comment:
these 2 appear to doing the exact same thing. prefer having only one:

`Emit(ctx context.Context, m Message)`
this lets us evolve the Message struct without breaking the api

Reply:
I mostly agree, although I could see a case being made for a variadic arg version for brevity:
```suggestion
	Emit(ctx context.Context, body []byte, attrKVs ...[]any) error
```
This version loses some compile time type safety just like a logger `With` API, but it may simplify callsites significantly:
```go
emitter.Emit(ctx, data, map[string]any{
	"key": "value",
	"field": 10,
})
```
```go
emitter.Emit(ctx, data, "key": "value", "field": 10)
```

Reply:
> emitter.Emit(ctx, data, "key", "value", "field", 10)

I can call it `SugarEmit` :-)

Reply:
That sugary syntax is supported for beholder message Attributes see here https://github.com/smartcontractkit/chainlink-common/pull/694/files#diff-d5bbc8388dc6b5c9b3b4011eb5cd2448839eb67b352eae919a8a68fb3c3fdb12R46-R47

Reply:
`Emit` variadic version https://github.com/smartcontractkit/chainlink-common/pull/694/commits/ec50c05c7b341745d81426767ff0a6e0f339a3d8

Reply:
IFIUC the consumer has to construct a `CustomMessage` for `EmitMessage`. Is there any value to consumers constructing a `CustomMessage` from the key value pairs they care about? If not, they may prefer to just put the key-value pairs into the API directly, which to me is a vote in favor for removing `EmitMessage` and keeping `Emit`

Reply:
Thats ok with me.
@jmank88 any thoughts on that ?

Reply:
@patrickhuie19, removed `EmitMessage` method, see [here](https://github.com/smartcontractkit/chainlink-common/pull/694/commits/1392fd20dad391512f255acf8a2c827d74b3d66c)

Reply:
I actually thought that `EmitMessage` was the more important method, since it would be able to evolve compatibly, and since a Message could be cumulatively constructed before being sent :shrug: 

What do the call sites for these methods actually look like?